timesketch/tsctl.py

```diff
@@ -1373,3 +1416,574 @@ def list_config():
+    and export context into a structured dictionary.
+    Args:
+        sketch: The timesketch.models.sketch.Sketch object to extract metadata from.
+    Returns:
+        A dictionary containing metadata about the sketch, including:
+            - Basic sketch info (ID, name, description, status, timestamps, owner).
+            - Permissions and sharing details.
+            - List of associated timelines with their details (including data sources).
+            - List of saved views (name, query, filter, DSL).
+            - List of stories (title, content).
+            - List of aggregations and aggregation groups.
+            - List of saved graphs.
+            - List of analysis sessions.
+            - List of DFIQ scenarios (including nested facets, questions, etc.).
+            - Sketch attributes.
+            - Export timestamp and Timesketch version.
+            - Comments.
+    """
+    # Schemas for marshalling, from the API resource mixin
+    schemas = ResourceMixin.fields_registry
+    print("Gathering metadata...")
+    metadata = {
+        "sketch_id": sketch.id,
+        "name": sketch.name,
+        "description": sketch.description,
+        "status": sketch.get_status.status,
+        "created_at": _isoformat_or_none(sketch.created_at),
+        "updated_at": _isoformat_or_none(sketch.updated_at),
+        "created_by": sketch.user.username if sketch.user else None,
+        "is_public": bool(sketch.is_public),
+        "labels": [label.label for label in sketch.labels],
+        "all_permissions": sketch.get_all_permissions(),
+        "timelines": [],
+        "views": [],
+        "stories": [],
+        "aggregations": [],
+        "aggregation_groups": [],
+        "graphs": [],
+        "analysis_sessions": [],
+        "scenarios": [],
+        "comments": [],
+        "attributes": api_utils.get_sketch_attributes(sketch),
+        "export_timestamp": datetime.datetime.now(datetime.timezone.utc).isoformat(),
+        "timesketch_version": version.get_version(),
+    }
+
+    # Timelines
+    for timeline in sketch.timelines:
+        marshalled_timeline = marshal(timeline, schemas["timeline"])
+        metadata["timelines"].append(marshalled_timeline)
+
+    # Views
+    for view in sketch.get_named_views:
+        marshalled_view = marshal(view, schemas["view"])
+        metadata["views"].append(marshalled_view)
+
+    # Stories
+    for story in sketch.stories:
+        marshalled_story = marshal(story, schemas["story"])
+        metadata["stories"].append(marshalled_story)
+
+    # Aggregations
+    for agg in sketch.aggregations:
+        marshalled_agg = marshal(agg, schemas["aggregation"])
+        metadata["aggregations"].append(marshalled_agg)
+
+    # Aggregation Groups
+    for group in sketch.aggregationgroups:
+        marshalled_group = marshal(group, schemas["aggregationgroup"])
+        metadata["aggregation_groups"].append(marshalled_group)
```

Comment:
This only stores the metadata info from the database like query used to aggregate but not the resulting images. Is this intentional?

Reply:
as discussed offline, we will not have the images for now, as with some other things that we might have missed, important that we document to the user what IS covered.